### PR TITLE
[PTX-2506] Added ScyallaDB app

### PR DESCRIPTION
**Pull request by vinayakshnd**
### ScyllaDB
1. Torpedo deploys `scylladb/scylla:4.0.0` as stateful set.
2. ScyllaDB is ready or not is determined based on output of `nodetool status`. In case of successful setup, below output is seen. Exec readiness probe uses script in `scylla` configMap to determine app readiness.
```
nodetool status
Datacenter: datacenter1
=======================
Status=Up/Down
|/ State=Normal/Leaving/Joining/Moving
--  Address        Load       Tokens       Owns (effective)  Host ID                               Rack
UN  10.233.127.67  356.04 KB  256          61.5%             b7250433-7f4b-414e-8632-d9b928f1fc4a  rack1
UN  10.233.76.19   351.15 KB  256          70.5%             814bf13f-8d4a-441d-bbba-c897c7441895  rack1
UN  10.233.121.53  359.24 KB  256          68.0%             013699ce-1fa8-4a32-86ec-640ce9ec9f6e  rack1
```
3. ScyllaDB forms a ring using a seed node. In our case, we use first pod `scylla-0.scylla._NAMESPACE_.svc.cluster.local` as seed node. Communication with other pods in statefulsets happen via headless service. Using just `scylla-0.scylla` does not work since the first pod when it tries to come up, it tries to connect with seed node (ie. with itself using `scylla-0.scylla`). But at this point since its not ready yet, it is not registered as endpoint to headless service and it fails to resolve it. Hence we need to use complete FQDN.

### Cassandra-stress
1. `cassandra-stress` tool is used to generate load. This runs as separate deployment and not as sidecar (to avoid stoping of IO because of node/pod fault injection by the test). 

2. Readiness of the tool is determined by at least 5 occurrences of stats lines in the output. In case of a working tool, below output is seen:
```
total,         17228,    6698,    6698,    6698,     7.3,     5.3,    26.0,    50.2,    55.8,    68.6,    5.0,  0.31657,      0,      0,       0,       0,       0,       0
total,         23735,    6507,    6507,    6507,     7.8,     6.6,    21.8,    34.7,    47.8,    52.0,    6.0,  0.25551,      0,      0,       0,       0,       0,       0
total,         29751,    6016,    6016,    6016,     8.2,     6.6,    27.0,    50.5,    65.2,    72.9,    7.0,  0.21335,      0,      0,       0,       0,       0,       0
total,         36340,    6589,    6589,    6589,     7.6,     6.6,    17.7,    25.1,    37.0,    37.5,    8.0,  0.18366,      0,      0,       0,       0,       0,       0
total,         43211,    6871,    6871,    6871,     7.2,     5.0,    25.9,    50.0,    68.7,    73.5,    9.0,  0.16231,      0,      0,       0,       0,       0,       0
total,         49814,    6603,    6603,    6603,     7.5,     3.8,    43.2,    63.8,    86.2,    87.2,   10.0,  0.14494,      0,      0,       0,       0,       0,       0
total,         56989,    7175,    7175,    7175,     7.0,     5.8,    17.6,    38.5,    54.4,    56.9,   11.0,  0.13144,      0,      0,       0,       0,       0,       0
```
3. Liveness of the tool/ IOs are not stuck is determined by change in number of occurances of stats lines in output.

4. Exec readiness and liveness probes uses script from `cassandra-stress-livenessprobe` and `cassandra-stress-livenessprobe` configMaps respectively.

### CODE REVIEW

**Comment by vinayakshnd** [comment]
Jenkins Job:
VolumeDriverDown with 5 mins between stop and start:
https://jenkins.portworx.co/job/Torpedo/job/tp-scylladb/2/consoleFull

**Comment by vinayakshnd** [comment]
VolumeDriverDown test able to catch customer issue with scyllaDB:
https://jenkins.portworx.co/job/Torpedo/job/tp-scylladb/12/console

**Comment by vinayakshnd** [comment]
On PX master build, VolumeDriverDown failed with [PWX-13598](https://portworx.atlassian.net/browse/PWX-13598)

Job: https://jenkins.portworx.co/job/Torpedo/job/tp-scylladb/18/

File: tests/basic/basic_test.go [hunk 1]
```diff
@@ -75,6 +75,11 @@ var _ = Describe("{VolumeDriverDown}", func() {
 						StopVolDriverAndWait([]node.Node{appNode})
 					})
 
+				Step(fmt.Sprintf("wait for %v minutes before starting volume driver",
+					Inst().ChaosLevel), func() {
+					time.Sleep(time.Duration(Inst().ChaosLevel) * time.Minute)
+				})
```
Comment (harsh-px) [inline tests/basic/basic_test.go:81]: remove chaos level

**Comment by harsh-px** [comment]
Need to retry with different timeouts between volume driver stop and start such that

1. timeout is as low as possible
2. we still reproduce the scylladb issue

**Pull request merged**